util/generate-ami-list.py

```diff
@@ -240,60 +250,35 @@ def get_all_aws_regions_from_ec2(region):
+    """Read the existing json file from the given path."""
+    with open(json_file_path) as json_file:
         # object_pairs_hook=OrderedDict allows to preserve input order
-        return json.load(cfn_file, object_pairs_hook=OrderedDict)
+        return json.load(json_file, object_pairs_hook=OrderedDict)
 
 
-def write_cfn_template(template_path, cfn_data):
-    """Write the CFN template represented by cfn_data to the given path."""
-    with open(template_path, "w") as cfn_file:
+def write_json_file(json_file_path, json_data):
+    """Write the json represented by json_data to the given path."""
+    with open(json_file_path, "w") as json_file:
         # setting separators to (',', ': ') to avoid trailing spaces after commas
-        json.dump(cfn_data, cfn_file, indent=2, separators=(",", ": "))
+        json.dump(json_data, json_file, indent=2, separators=(",", ": "))
         # add new line at the end of the file
-        cfn_file.write("\n")
-
-
-def update_cfn_template(cfn_template_file, amis_to_update):
-    """Update in-place the mappings section of cfn_template_file with the AMI IDs contained in amis_to_update."""
-    # Read in existing CFN template
-    cfn_data = read_cfn_template(cfn_template_file)
-    # update id for new amis without removing regions that are not in the amis_to_update dict
-    for mapping_name in ARCHITECTURES_TO_MAPPING_NAME.values():
-        current_amis_for_mapping = cfn_data.get("Mappings").get(mapping_name, {})
-        for region, amis_to_update_region_mapping in amis_to_update.get(mapping_name, {}).items():
-            if not amis_to_update_region_mapping:
-                # No new AMIs for this region in this mapping
-                continue
-            elif region not in current_amis_for_mapping and amis_to_update_region_mapping:
-                # There are no AMIs for this region in the existing mapping, but there are in the updated version
-                current_amis_for_mapping[region] = amis_to_update_region_mapping
-            else:
-                current_amis_for_mapping[region].update(amis_to_update_region_mapping)
-            current_amis_for_mapping[region] = OrderedDict(sorted(current_amis_for_mapping[region].items()))
-
-        # enforce alphabetical regions order
-        current_amis_for_mapping = OrderedDict(sorted(current_amis_for_mapping.items()))
-        cfn_data.get("Mappings")[mapping_name] = current_amis_for_mapping
-    # Ensure mappings are sorted
-    cfn_data["Mappings"] = OrderedDict(sorted(cfn_data["Mappings"].items()))
-    # Write back modified CFN template
-    write_cfn_template(cfn_template_file, cfn_data)
-
-
-def update_amis_txt(amis_txt_file, cfn_template_file):
-    """Write amis_txt_file using the updated information contained in cfn_template_file."""
-    cfn_data = read_cfn_template(cfn_template_file)
-    amis_txt = convert_json_to_txt(cfn_data.get("Mappings"))
+        json_file.write("\n")
+
+
+def write_amis_txt(amis_txt_file, json_file):
```

Comment:
I think this needs to be remover as well. If mapping was removed from template also the ami.txt must be removed